doc/src/manual/functions.md

```diff
@@ -202,39 +205,53 @@ julia> function (x)
 #3 (generic function with 1 method)
 ```
 
-This creates a function taking one argument `x` and returning the value of the polynomial `x^2 +
-2x - 1` at that value. Notice that the result is a generic function, but with a compiler-generated
-name based on consecutive numbering.
+This creates a function taking one argument `x` and returning the value of the polynomial
+`x^2 + 2x - 1` at that value. Notice that the result is a generic function, but with a
+compiler-generated name based on consecutive numbering. (Neither of these can really be used
+as-is; these syntaxes are for creating anonymous functions to be passed to other functions.)
 
 The primary use for anonymous functions is passing them to functions which take other functions
 as arguments. A classic example is [`map`](@ref), which applies a function to each value of
 an array and returns a new array containing the resulting values:
 
 ```jldoctest
-julia> map(round, [1.2,3.5,1.7])
+julia> map(round, [1.2, 3.5, 1.7])
 3-element Array{Float64,1}:
  1.0
  4.0
  2.0
 ```
 
-This is fine if a named function effecting the transform already exists to pass as the first argument
-to [`map`](@ref). Often, however, a ready-to-use, named function does not exist. In these
-situations, the anonymous function construct allows easy creation of a single-use function object
-without needing a name:
+This is fine if you already have a suitable named function to pass as the first argument to
+[`map`](@ref). However, if you need to do a one-off transformation and don't already have an
+appropriate function, you can easily create an anonymous function to do the job:
 
 ```jldoctest
-julia> map(x -> x^2 + 2x - 1, [1,3,-1])
+julia> map(x -> x^2 + 2x - 1, [1, 3,.-1])
 3-element Array{Int64,1}:
   2
  14
  -2
 ```
 
-An anonymous function accepting multiple arguments can be written using the syntax `(x,y,z)->2x+y-z`.
-A zero-argument anonymous function is written as `()->3`. The idea of a function with no arguments
-may seem strange, but is useful for "delaying" a computation. In this usage, a block of code is
-wrapped in a zero-argument function, which is later invoked by calling it as `f`.
+It is also possible to do this using the second syntax:
+
+```jldoctest
```

Comment:
I think giving every example with both syntaxes is confusing. The `->` syntax is preferred, so it should be used here and the other syntax can be mentioned later.